bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/db/JdbcBaseDAO.java

```diff
@@ -703,46 +707,65 @@ protected State objectAsState(Item item, @Nullable Unit<? extends Quantity<?>> u
     protected ZonedDateTime objectAsZonedDateTime(Object v) {
         if (v instanceof Long) {
             return ZonedDateTime.ofInstant(Instant.ofEpochMilli(((Number) v).longValue()), ZoneId.systemDefault());
-        } else if (v instanceof java.sql.Date) {
-            return ZonedDateTime.ofInstant(Instant.ofEpochMilli(((java.sql.Date) v).getTime()), ZoneId.systemDefault());
-        } else if (v instanceof LocalDateTime) {
-            return ((LocalDateTime) v).atZone(ZoneId.systemDefault());
-        } else if (v instanceof Instant) {
-            return ((Instant) v).atZone(ZoneId.systemDefault());
-        } else if (v instanceof java.sql.Timestamp) {
-            return ((java.sql.Timestamp) v).toInstant().atZone(ZoneId.systemDefault());
-        } else if (v instanceof java.lang.String) {
-            return ZonedDateTime.ofInstant(java.sql.Timestamp.valueOf(v.toString()).toInstant(),
+        } else if (v instanceof java.sql.Date objectAsDate) {
+            return ZonedDateTime.ofInstant(Instant.ofEpochMilli(objectAsDate.getTime()), ZoneId.systemDefault());
+        } else if (v instanceof LocalDateTime objectAsLocalDateTime) {
+            return objectAsLocalDateTime.atZone(ZoneId.systemDefault());
+        } else if (v instanceof Instant objectAsInstant) {
+            return objectAsInstant.atZone(ZoneId.systemDefault());
+        } else if (v instanceof java.sql.Timestamp objectAsTimestamp) {
+            return objectAsTimestamp.toInstant().atZone(ZoneId.systemDefault());
+        } else if (v instanceof java.lang.String objectAsString) {
+            return ZonedDateTime.ofInstant(java.sql.Timestamp.valueOf(objectAsString).toInstant(),
                     ZoneId.systemDefault());
         }
-        throw new UnsupportedOperationException("Date of type " + v.getClass().getName() + " is not supported");
+        throw new UnsupportedOperationException("Date of type '" + v.getClass().getName() + "' is not supported");
     }
 
     protected Integer objectAsInteger(Object v) {
         if (v instanceof Byte) {
             return ((Byte) v).intValue();
+        } else if (v instanceof Integer) {
+            return (Integer) v;
```

Comment:
You can a variable to avoid the cast (like you did in next methods.

Reply:
That results in this SAT error:
```
[ERROR] org.openhab.persistence.jdbc.internal.db.JdbcBaseDAO.java:[728]
Suspicious comparison of Integer references in org.openhab.persistence.jdbc.internal.db.JdbcBaseDAO.objectAsInteger(Object)
```

I was not able to figure out why - to me it seems like a false positive, so I decided to leave it for now. I would welcome any tips. 🙂

Just to be clear - this version would generate two of such errors:
```java
    protected Integer objectAsInteger(Object v) {
        if (v instanceof Byte objectAsByte) {
            return objectAsByte.intValue();
        } else if (v instanceof Integer objectAsInteger) {
            return objectAsInteger;
        }
        throw new UnsupportedOperationException("Integer of type '" + v.getClass().getName() + "' is not supported");
    }
```

Reply:
Strange. But let's merge as it is.